week 2/script.js

```diff
@@ -0,0 +1,232 @@
+    rates: {
+        USD: 1.23396
+    }
+};
+
+class CurrencyRate {
+    constructor(code, rate, date) {
+        this.code = code;
+        this.rate = rate;
+        this.date = date;
+    }
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+    var coll = document.querySelectorAll(".collapsible");
+    coll.forEach(function(collapsible) {
+        collapsible.addEventListener("click", function() {
+            this.classList.toggle("active");
+            var content = this.nextElementSibling;
+            if (content.style.display === "block") {
+                content.style.display = "none";
+            } else {
+                content.style.display = "block";
+            }
+        });
+    });
+
+    document.querySelector('#insertCurrencyForm').addEventListener('submit', function(event) {
+        event.preventDefault();
+        insertCurrencyRate();
+    });
+
+    document.querySelector('#convertCurrencyForm').addEventListener('submit', function(event) {
+        event.preventDefault();
+        convertCurrency();
+    });
+
+    document.querySelector('#updateCurrencyForm').addEventListener('submit', function(event) {
+        event.preventDefault();
+        updateRate();
+    });
+
+    populateCurrencyDropdown();
+});
+
+
+let currencyRatesArray = [];
+currencyRatesArray.push(new CurrencyRate("EUR", 1.0, currencyRates.date));
+
+currencyRatesArray = currencyRatesArray.concat(
+    Object.entries(currencyRates.rates).map(
+        ([code, rate]) => new CurrencyRate(code, rate, currencyRates.date)
+    )
+);
+
+function insertCurrencyRate() {
+    const base = document.querySelector('#base').value.toUpperCase();
+    const targetCurrency = document.querySelector('#targetCurrency').value.trim().toUpperCase();
+    const rate = parseFloat(document.querySelector('#rate').value);
+
+    if (!currencyRates) {
+        currencyRates = {
+            timestamp: Date.now(),
+            base: base,
+            date: new Date().toISOString().split('T')[0],
+            rates: {}
+        };
+    }
+
+    if (targetCurrency === '' || rate === '') {
+        alert('Currency or Rate cannot be blank');
+        return;
+    }
+
+    if (currencyRates.base !== base) {
+        alert('Base currency mismatch! Current base is ' + currencyRates.base);
+        return;
+    }
+
+    if (currencyRatesArray.length > 0 && currencyRatesArray.findIndex(obj => obj.code == targetCurrency) !== -1) {
+        alert('Currency already exists in the list');
+        return;
+    }
+
+    currencyRates.timestamp = Date.now();
+    currencyRates.date = new Date().toISOString().split('T')[0];
+    currencyRates.rates[targetCurrency] = rate;
+    document.querySelector('#newRateOutput').textContent = JSON.stringify(currencyRates, null, 2);
+
+    currencyRatesArray.push(new CurrencyRate(targetCurrency, rate, currencyRates.date));
+    populateCurrencyDropdown();
+}
+
+function populateCurrencyDropdown() {
+    startTimer();
```

Comment:
The call to `startTimer()` does not belong here; it is unrelated to populating currency dropdowns. I would move it to the `DOMContentLoaded` callback, line 19.

Reply:
Done